cmd/renterd/config.go

```diff
@@ -0,0 +1,342 @@
+// set via environment variable or config file.
+func setSeedPhrase() {
+	// retry until a valid seed phrase is entered
+	for {
+		fmt.Println("")
+		fmt.Println("Type in your 12-word seed phrase and press enter. If you do not have a seed phrase yet, type 'seed' to generate one.")
+		phrase, err := readPasswordInput("Enter seed phrase")
+		if err != nil {
+			stdoutFatalError("Could not read seed phrase: " + err.Error())
+		}
+
+		if strings.ToLower(strings.TrimSpace(phrase)) == "seed" {
+			// generate a new seed phrase
+			var seed [32]byte
+			phrase = wallet.NewSeedPhrase()
+			if err := wallet.SeedFromPhrase(&seed, phrase); err != nil {
+				panic(err)
+			}
+			key := wallet.KeyFromSeed(&seed, 0)
+			fmt.Println("")
+			fmt.Println("A new seed phrase has been generated below. " + wrapANSI("\033[1m", "Write it down and keep it safe.", "\033[0m"))
+			fmt.Println("Your seed phrase is the only way to recover your Siacoin. If you lose your seed phrase, you will also lose your Siacoin.")
+			fmt.Println("")
+			fmt.Println(wrapANSI("\033[34;1m", "Seed Phrase:", "\033[0m"), phrase)
+			fmt.Println(wrapANSI("\033[34;1m", "Wallet Address:", "\033[0m"), types.StandardUnlockHash(key.PublicKey()))
+
+			// confirm seed phrase
+			for {
+				fmt.Println("")
+				fmt.Println(wrapANSI("\033[1m", "Please confirm your seed phrase to continue.", "\033[0m"))
+				confirmPhrase, err := readPasswordInput("Enter seed phrase")
+				if err != nil {
+					stdoutFatalError("Could not read seed phrase: " + err.Error())
+				} else if confirmPhrase == phrase {
+					cfg.Seed = phrase
+					return
+				}
+
+				fmt.Println(wrapANSI("\033[31m", "Seed phrases do not match!", "\033[0m"))
+				fmt.Println("You entered:", confirmPhrase)
+				fmt.Println("Actual phrase:", phrase)
+			}
+		}
+
+		var seed [32]byte
+		err = wallet.SeedFromPhrase(&seed, phrase)
+		if err == nil {
+			// valid seed phrase
+			cfg.Seed = phrase
+			return
+		}
+		fmt.Println(wrapANSI("\033[31m", "Invalid seed phrase:", "\033[0m"), err)
+		fmt.Println("You entered:", phrase)
+	}
+}
+
+// setAPIPassword prompts the user to enter an API password if one is not
+// already set via environment variable or config file.
+func setAPIPassword() {
+	// retry until a valid API password is entered
+	for {
+		fmt.Println("Please choose a password for the renterd admin UI.")
+		fmt.Println("This password will be required to access the admin UI in your web browser.")
+		fmt.Println("(The password must be at least 4 characters.)")
+		var err error
+		cfg.HTTP.Password, err = readPasswordInput("Enter password")
+		if err != nil {
+			stdoutFatalError("Could not read password:" + err.Error())
+		}
+
+		if len(cfg.HTTP.Password) >= 4 {
+			break
+		}
+
+		fmt.Println(wrapANSI("\033[31m", "Password must be at least 4 characters!", "\033[0m"))
+		fmt.Println("")
+	}
+}
+
+func setAdvancedConfig() {
+	input, err := readInput("Would you like to configure advanced settings? (yes/no)")
```

Comment:
`nit` we repeat this pattern a couple of times: ask yes/no question read input and check for `yes` - could be extracted into a helper function